### PR TITLE
script downloads structure of the book, creates directory tree of the…

### PR DESCRIPTION
**Pull request by worldmind**
… book, downloads chapters and creates data files

### CODE REVIEW

File: scripts/download_from_lesswrong.com.py [hunk 1]
```diff
@@ -1,2 +1,130 @@
...
+
+                for post in chapter["posts"]:
+                    post_dirname = (
+                        str(post_i).zfill(2) + "_" + safe_dirname(post["title"])
+                    )
+                    post_p = Path(sequence_p.absolute(), post_dirname)
+
+                    try:
+                        post_p.mkdir(parents=True, exist_ok=True)
+                    except Exception as err:
```
Comment (worldmind) [inline scripts/download_from_lesswrong.com.py:99]: same question about exceptions, perhaps info logging is enough

File: scripts/download_from_lesswrong.com.py [hunk 2]
```diff
@@ -1,2 +1,130 @@
...
+            post_i = 1
+            if "contents" in sequence and isinstance(sequence["contents"], dict):
+
+                intro_post = {
+                    "title": "Intro",
+                    "contents": {"html": sequence["contents"]["html"]},
+                    "modifiedAt": sequence["contents"]["editedAt"],
+                    "_id": sequence["_id"],
+                    "user": sequence["user"],
+                    "pageUrl": "https://www.lesswrong.com/s/" + sequence["_id"],
```
Comment (worldmind) [inline scripts/download_from_lesswrong.com.py:84]: Do they have info about authors? Or it's in body?

File: scripts/download_from_lesswrong.com.py [hunk 3]
```diff
@@ -1,2 +1,130 @@
...
+                    if post_i == 0:
+                        write_post(post_p.absolute(), post)
+                    else:
+                        posts_for_download[post["_id"]] = post_p.absolute()
+
+                    post_i += 1
+
+    log.info("Directory tree created at '%s'", p_base.absolute())
+
+    for id in list(posts_for_download.keys()):
```
Comment (worldmind) [inline scripts/download_from_lesswrong.com.py:114]: `list` is not needed

File: scripts/download_from_lesswrong.com.py [hunk 4]
```diff
@@ -1,2 +1,130 @@
...
+        post = lw.download_post(id)
+
+        if "error" in post:
+            log.error("Post _id='%s' not downloaded: %s", id, post["error"])
+        else:
+            write_post(posts_for_download[id], post["data"])
+
+        time.sleep(1)
+
+    return
```
Comment (worldmind) [inline scripts/download_from_lesswrong.com.py:125]: not needed

**Review by worldmind — COMMENTED** [review]
@yozhio looks really good, added some nitpicks, up to you what to change

File: scripts/download_from_lesswrong.com.py (hunk 2 above)
Comment (yozhio) [reply scripts/download_from_lesswrong.com.py]: It's "user"

File: scripts/lib/lesswrong.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+{result {title version books {title sequences {_id title user {displayName} \
+contents {editedAt html} chapters {posts {title _id}}}}}}}'
+
+API_POST_QUERY = '{post(input:{selector:{_id:"POST_ID"}}) \
+{result{_id title user{displayName}modifiedAt pageUrl contents{html}tags{name}}}}'
+
+###
+def download_collection(slug: str) -> dict:
+    payload = {"query": None, "variables": None, "operationName": None}
+    payload["query"] = API_COLLECTION_QUERY.replace("C_SLUG", slug)
```
[the remark this answers is not in the document]
Comment (yozhio) [reply scripts/lib/lesswrong.py:25]: i am getting exception because of braces in json

File: scripts/download_from_lesswrong.com.py (hunk 1 above)
Comment (yozhio) [reply scripts/download_from_lesswrong.com.py:60]: sorry, i didn't catch an idea. you mean, that i don't need to use try/except blocks?

Comment (worldmind) [reply scripts/download_from_lesswrong.com.py:60]: Yes, result will be same - script will exit, just need to check that last log message has enough info

File: scripts/download_from_lesswrong.com.py (hunk 2 above)
Comment (worldmind) [reply scripts/download_from_lesswrong.com.py]: Wondering do they have cases when it's more than one person

Comment (yozhio) [reply scripts/download_from_lesswrong.com.py]: i’ll look later

**Comment by worldmind** [comment]
After adding all author-related fields to meta please run
```
git rebase -i 4e6b52c893e0d31ab26aa6ba92a6ecbe0743a193
```
and mark all your commits exclude first as `fixup` instead of `pick`, all commits will be squashed, no need to have commits with fixes in history

**Comment by worldmind** [comment]
Cool, thank you, did small changes mostly for make linter happy, downloaded everything and also commited/pushed.

**Pull request merged**